### PR TITLE
Add more documentation

### PR DESCRIPTION
**Pull request by noelwelsh**
This is, I think, enough documentation to start with. There could be more documentation, but I don't think it's necessary to hold up a release waiting for it.

Also improves the `Layout` API a little bit.

### CODE REVIEW

File: docs/src/pages/creating-visualizations.md [hunk 1]
```diff
@@ -1,2 +1,84 @@
 # Creating Visualizations
 
+Visualization in Chartreuse are built out of small components. The most important components are:
```
Comment (danielost) [inline docs/src/pages/creating-visualizations.md]: I think there's a typo in **_Visualization_** -> **_Visualizations_**.

File: docs/src/pages/creating-visualizations.md [hunk 2]
```diff
@@ -1,2 +1,84 @@
...
+
+```scala mdoc:silent
+import doodle.core.Color
+
+layout
+  .forThemeable(themeable => themeable.withStrokeColor(Themeable.Override(Some(Color.chartreuse))))
+```
+
+Builder methods *always* return a modified copy of the object they're called on, 
+so it's always safe to a call a builder method even if you used a component in another place.
```
Comment (danielost) [inline docs/src/pages/creating-visualizations.md]: Unnecessary _**a**_ before **_call_**.

File: docs/src/pages/creating-visualizations.md [hunk 3]
```diff
@@ -1,2 +1,84 @@
...
+This is a core part of Chartreuse's design philosophy, as described in [Core Concepts](concepts.md)
+
+In the example above we used `toLayer` and `toPlot` to convert types. These are convenience methods.
+You can, for example, construct a `Plot` by calling it's constructor but it's much simpler to type `.` and follow the auto-complete to turn a `Layer` into a `Plot`.
```
Comment (danielost) [inline docs/src/pages/creating-visualizations.md]: Replace **_it's constructor_** with **_its constructor_**.

**Review by danielost — CHANGES_REQUESTED** [review]

File: docs/src/pages/creating-visualizations.md (hunk 3 above)
Comment (noelwelsh) [reply docs/src/pages/creating-visualizations.md]: Thanks! I've fixed those typos and improved the wording in a few other cases.

Comment (danielost) [reply docs/src/pages/creating-visualizations.md]: That's great!

**Pull request merged**